### DISCUSSION TITLE
Simple command output doesn't appear in Obsidian note

### DISCUSSION THREAD

**Discussion by losteris** [Taitava/obsidian-shellcommands#410] [Debugging and testing]
### Obsidian debug information

Simple command output doesn't appear in Obsidian note. I tried `ls` without the backticks.

SYSTEM INFO:
	Obsidian version: v1.5.12
	Installer version: v1.5.12
	Operating system: MacOS M1 Darwin Kernel ARM64_T8103 23.3.0
	Live preview: on
	Base theme: dark
	Community theme: Obsidian Nord v0.2.0
	Snippets enabled: 0
	Restricted mode: off
	Plugins installed: 16
	Plugins enabled: 14
		1: Tag Wrangler v0.6.1
		2: Calendar v1.5.10
		3: Dataview v0.5.66
		4: Wordy v1.3.1
		5: Ollama v0.0.1
		6: Regex Find/Replace v1.2.0
		7: Advanced URI v1.40.1
		8: Templater v2.2.3
		9: Sort & Permute lines v0.7.0
		10: Git v2.24.1
		11: Iconize v2.12.0
		12: Global Search and Replace v0.3.0
		13: Pandoc Plugin v0.4.1
		14: Shell commands v0.22.0

### What shell are you using?

Bash

### Please describe the problem

I can't get output from a simple ls command.

**Comment by Taitava** [comment]
Hi and thank you for your message! 🙂 

Did you try to [change the shell command's output channel](https://publish.obsidian.md/shellcommands/Output+handling/Output+channels#How+to+select+output+channels+for+a+shell+command) from _Ignore_ to [Current file: caret position](https://publish.obsidian.md/shellcommands/Output+handling/Output+channel+-+Current+file)? By default, all output to _standard output_ (`stdout`) is ignored.

**Reply by losteris** [reply]
Ahh! I was able to get that to work by creating a specific shell command. Thank you!

I was thinking that I would be able to type whatever shell command I wanted in Obsidian, highlight the command, and execute the command via command palette keyboard shortcut, and stdout would appear after the highlighted command.

Is that a capability I'm just missing in the documentation?

**Reply by Taitava** [reply]
It's probably not very well documented, but you can create this kind of simple shell command:
```bash
{{!selection}}
```

This executes whatever text happens to be selected in a note in Obsidian as a shell command. The exclamation mark `!` is important as it tells the SC plugin to keep all special characters (e.g. `"'-`) as-is, [without escaping](https://publish.obsidian.md/shellcommands/Variables/Escaping+special+characters+in+variable+values#Prevent+escaping+-+but+only+after+deep+consideration) them.
